judge/judgedaemon.main.php

```diff
@@ -872,6 +877,50 @@ function fetch_executable_internal(
         continue;
     }
 
+    if ($type == 'output_visualization') {
+        if ($lastWorkdir !== null) {
+            cleanup_judging($lastWorkdir);
+            $lastWorkdir = null;
+        }
+        foreach ($row as $judgeTask) {
+            if (isset($judgeTask['output_visualizer_script_id'])) {
+                // Visualization of output which this host judged requested.
+                $run_config = dj_json_decode($judgeTask['run_config']);
+                $tmpfile = tempnam(TMPDIR, 'output_visualization_');
+                [$runpath, $error] = fetch_executable_internal(
+                    $workdirpath,
+                    'output_visualization',
+                    $judgeTask['output_visualizer_script_id'],
+                    $run_config['hash']
+                );
+                if (isset($error)) {
+                    // FIXME
+                    continue;
+                }
+
+                $teamoutput = $workdir . "/testcase" . sprintf('%05d', $judgeTask['testcase_id']) . '/1/program.out';
+                $visual_cmd = implode(' ', array_map('dj_escapeshellarg',
+                    [$runpath, $teamoutput, $tmpfile]));
+                system($visual_cmd, $retval);
+                if ($retval !== 0) {
+                    error("Running '$runpath' failed.");
```

Comment:
let's rather report an internal error like we do in other cases (and not crash judgedaemons)